bridges/ethMultiversX/bridgeExecutor.go

```diff
@@ -587,56 +581,42 @@ func (executor *bridgeExecutor) isMintBurnOnEthereum(ctx context.Context, erc20A
 	return isMintBurn
 }
 
-func (executor *bridgeExecutor) isMintBurnOnMultiversX(ctx context.Context, erc20Address common.Address) bool {
+func (executor *bridgeExecutor) isMintBurnOnMultiversX(ctx context.Context, token []byte) bool {
 
-	isMintBurn, err := executor.multiversXClient.IsMintBurnAllowed(ctx, erc20Address.Bytes())
+	isMintBurn, err := executor.multiversXClient.IsMintBurnAllowed(ctx, token)
 	if err != nil {
 		return false
 	}
 	return isMintBurn
 }
 
-func (executor *bridgeExecutor) checkAvailableTokens(ctx context.Context, tokens []common.Address, amounts []*big.Int) error {
-	transfers := executor.getCumulatedTransfers(tokens, amounts)
+func (executor *bridgeExecutor) checkAvailableTokensOnEthereum(ctx context.Context, tokens []common.Address, convertedTokens [][]byte, amounts []*big.Int) error {
+	tokens, convertedTokens, amounts = executor.getCumulatedTransfers(tokens, convertedTokens, amounts)
 
-	return executor.checkCumulatedTransfers(ctx, transfers)
+	return executor.checkCumulatedTransfers(ctx, tokens, convertedTokens, amounts)
 }
 
-func (executor *bridgeExecutor) extractList(batch *clients.TransferBatch) (*ArgListsBatch, error) {
-	arg := ArgListsBatch{}
-
-	for _, dt := range batch.Deposits {
-		recipient := common.BytesToAddress(dt.ToBytes)
-		arg.Recipients = append(arg.Recipients, recipient)
-
-		token := common.BytesToAddress(dt.ConvertedTokenBytes)
-		arg.Tokens = append(arg.Tokens, token)
-
-		amount := big.NewInt(0).Set(dt.Amount)
-		arg.Amounts = append(arg.Amounts, amount)
-
-		nonce := big.NewInt(0).SetUint64(dt.Nonce)
-		arg.Nonces = append(arg.Nonces, nonce)
+func (executor *bridgeExecutor) getCumulatedTransfers(tokens []common.Address, convertedTokens [][]byte, amounts []*big.Int) ([]common.Address, [][]byte, []*big.Int) {
+	cumulatedAmounts := make(map[common.Address]*big.Int)
+	uniqueTokens := make([]common.Address, 0)
+	uniqueConvertedTokens := make([][]byte, 0)
 
-		arg.ConvertedTokenBytes = append(arg.ConvertedTokenBytes, dt.ConvertedTokenBytes)
-	}
-
-	return &arg, nil
-}
-
-func (executor *bridgeExecutor) getCumulatedTransfers(tokens []common.Address, amounts []*big.Int) map[common.Address]*big.Int {
-	transfers := make(map[common.Address]*big.Int)
 	for i, token := range tokens {
-		existing, found := transfers[token]
-		if !found {
-			existing = big.NewInt(0)
-			transfers[token] = existing
+		if _, exists := cumulatedAmounts[token]; exists {
```

Comment:
The code reads worse in this changeset. How about this:
```GO
for i, token := range tokens {
    existingValue, exists := cumulatedAmounts[token]
    if exists {
        existingValue.Add(existingValue, amounts[i])
        continue
    }

    cumulatedAmounts[token] = amounts[i]
    uniqueTokens = append(uniqueTokens, token)
    uniqueConvertedTokens = append(uniqueConvertedTokens, convertedTokens[i])

}


```

Reply:
done 👍